tests/components/notify/test_notify_targets.py

```diff
@@ -0,0 +1,85 @@
+"""The tests for notify services that change targets."""
+import unittest
+
+from homeassistant.components import notify
+from homeassistant.core import HomeAssistant
+
+
+async def test_same_targets(hass: HomeAssistant):
+    """Test not changing the targets in a notify service."""
+    test = TestNotificationService(hass)
+    await test.async_setup(hass, "notify", "test")
+    await test.async_register_services()
+    await hass.async_block_till_done()
+
+    assert hasattr(test, "registered_targets")
+    assert test.registered_targets == {"test_a": 1, "test_b": 2}
+
+    await test.async_register_services()
+    await hass.async_block_till_done()
+    assert test.registered_targets == {"test_a": 1, "test_b": 2}
+
+
+async def test_change_targets(hass: HomeAssistant):
+    """Test changing the targets in a notify service."""
+    test = TestNotificationService(hass)
+    await test.async_setup(hass, "notify", "test")
+    await test.async_register_services()
+    await hass.async_block_till_done()
+
+    assert hasattr(test, "registered_targets")
+    assert test.registered_targets == {"test_a": 1, "test_b": 2}
+
+    test.target_list = {"a": 0}
+    await test.async_register_services()
+    await hass.async_block_till_done()
+    assert test.target_list == {"a": 0}
+    assert test.registered_targets == {"test_a": 0}
+
+
+async def test_add_targets(hass: HomeAssistant):
+    """Test adding the targets in a notify service."""
+    test = TestNotificationService(hass)
+    await test.async_setup(hass, "notify", "test")
+    await test.async_register_services()
+    await hass.async_block_till_done()
+
+    assert hasattr(test, "registered_targets")
+    assert test.registered_targets == {"test_a": 1, "test_b": 2}
+
+    test.target_list = {"a": 1, "b": 2, "c": 3}
+    await test.async_register_services()
+    await hass.async_block_till_done()
+    assert test.target_list == {"a": 1, "b": 2, "c": 3}
+    assert test.registered_targets == {"test_a": 1, "test_b": 2, "test_c": 3}
+
+
+async def test_remove_targets(hass: HomeAssistant):
+    """Test removing targets from the targets in a notify service."""
+    test = TestNotificationService(hass)
+    await test.async_setup(hass, "notify", "test")
+    await test.async_register_services()
+    await hass.async_block_till_done()
+
+    assert hasattr(test, "registered_targets")
+    assert test.registered_targets == {"test_a": 1, "test_b": 2}
+
+    test.target_list = {"c": 1}
+    await test.async_register_services()
+    await hass.async_block_till_done()
+    assert test.target_list == {"c": 1}
+    assert test.registered_targets == {"test_c": 1}
+
+
+class TestNotificationService(notify.BaseNotificationService, unittest.TestCase):
```

Comment:
We don't need to inherit TestCase.
```suggestion
class TestNotificationService(notify.BaseNotificationService):
```

Reply:
I get a 'cannot collect class because it has a __init__ constructor' warning when I remove the test class. Is that acceptable?

Reply:
Rename the class and don't prefix it with `Test`.